src/biokbase/narrative/jobs/appmanager.py

```diff
@@ -768,15 +768,15 @@ def _map_inputs(self, input_mapping, params, spec_params):
                     # This is case when slashes in target_prop separeate
                     # elements in nested maps. We ignore escaped slashes
                     # (separate backslashes should be escaped as well).
-                    bck_slash = u"\u244A"
-                    fwd_slash = u"\u20EB"
+                    bck_slash = "\u244A"
+                    fwd_slash = "\u20EB"
                     temp_string = target_prop.replace("\\\\", bck_slash)
                     temp_string = temp_string.replace("\\/", fwd_slash)
                     temp_path = []
                     for part in temp_string.split("/"):
                         part = part.replace(bck_slash, "\\")
                         part = part.replace(fwd_slash, "/")
-                        temp_path.append(part.encode('ascii', 'ignore'))
+                        temp_path.append(part.encode('ascii', 'ignore').decode("ascii"))
```

Comment:
Do we want to strip non-ascii chars or should we support them now?

Reply:
That's all an ugly bit of parsing to handle some side cases that I don't think are documented anywhere, and I'm really not sure anything uses.

Short version = the `target_property` field in an app spec can be set to something like "foo/bar/baz", and it'll put the value in a dictionary like {foo: {bar: {baz: <value>}}}. That weirdness up there handles including slashes in the name, especially escaped slashes. So a `target_property` of "foo/bar\/baz" should be {foo: {bar/baz: <value>}}.

Whoever wrote this bit (Roman, I think), did some temporary things by turning slashes into their unicode values, then used the `encode` function to restore them. In Python 3, `encode` always returns a `bytes` object, so we have to decode that (again) to make it a useful string.

For that purpose, everything always supported ascii, and would fail at the Catalog service anyway. (er, probably. Either way, it's expected that the app specs this code sees are up to snuff).